src/Akka.Streams.ScanMemory/Program.cs

```diff
@@ -14,22 +15,25 @@ static async Task Main(string[] args)
         {
             var system = ActorSystem.Create("MySystem");
             var streamBits = Source.ActorRef<int>(1000, OverflowStrategy.DropHead)
+                .Select(x => ThreadLocalRandom.Current.Next(1, 10000))
                 .GroupedWithin(100, TimeSpan.FromMilliseconds(100))
                 .Scan(new SortedSet<int>(), (set, ints) =>
                 {
                     foreach (var i in ints)
                         set.Add(i);
 
                     return set;
-                }).PreMaterialize(system.Materializer());
+                })
+                .Select(x => x.ToImmutableSortedSet()) // force copy into an immutable set so it's safe to pass between actors
```

Comment:
Force us to copy the `SortedSet<int>` into an `ImmutableSortedSet<int>` each time - if we were going to send the `Scan` results off to another actor this is what we'd need to do anyway in order to avoid the type of side effects I describe in https://petabridge.com/blog/how-to-design-akkadotnet-domain-messages/